applications/asset_tracker_v2/tests/debug_module/src/debug_module_test.c

```diff
@@ -0,0 +1,236 @@
+ * infuse callback events directly from the test runner.
+ */
+watchdog_evt_handler_t debug_module_watchdog_callback;
+
+/* Suite teardown finalizes with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+void setUp(void)
+{
+	mock_watchdog_app_Init();
+	mock_modules_common_Init();
+	mock_event_manager_Init();
+}
+
+void tearDown(void)
+{
+	mock_watchdog_app_Verify();
+	mock_modules_common_Verify();
+	mock_event_manager_Verify();
+}
+
+static void latch_watchdog_callback(watchdog_evt_handler_t handler, int calls)
+{
+	debug_module_watchdog_callback = handler;
+}
+
+static void validate_debug_data_ready_evt(struct event_header *eh, int no_of_calls)
+{
+	struct debug_module_event *event = cast_debug_module_event(eh);
+
+	TEST_ASSERT_EQUAL(DEBUG_EVT_MEMFAULT_DATA_READY, event->type);
+}
+
+void setup_debug_module_in_init_state(void)
+{
+	struct app_module_event *app_module_event = new_app_module_event();
+
+	app_module_event->type = APP_EVT_START;
+
+	static struct module_data expected_module_data = {
+		.name = "debug",
+		.msg_q = NULL,
+		.supports_shutdown = true,
+	};
+
+	__wrap_watchdog_register_handler_ExpectAnyArgs();
+	__wrap_watchdog_register_handler_AddCallback(&latch_watchdog_callback);
+	__wrap_module_start_ExpectAndReturn(&expected_module_data, 0);
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)app_module_event));
+
+	k_free(app_module_event);
+}
+
+void test_memfault_trigger_metric_sampling_on_gps_fix(void)
+{
+	setup_debug_module_in_init_state();
+
+	/* Update this function to expect the search time and number of satellites. */
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsTimeToFix),
+						60000,
+						0);
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsSatellitesTracked),
+						4,
+						0);
+	__wrap_memfault_metrics_heartbeat_debug_trigger_Ignore();
+
+	struct gps_module_event *gps_module_event = new_gps_module_event();
+
+	gps_module_event->type = GPS_EVT_DATA_READY;
+	gps_module_event->data.gps.satellites_tracked = 4;
+	gps_module_event->data.gps.search_time = 60000;
+
+	TEST_ASSERT_EQUAL(0, DEBUG_MODULE_EVT_HANDLER((struct event_header *)gps_module_event));
+
+	k_free(gps_module_event);
+}
+
+void test_memfault_trigger_metric_sampling_on_gps_timeout(void)
+{
+	resetTest();
+	setup_debug_module_in_init_state();
+
+	/* Update this function to expect the search time and number of satellites. */
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsTimeoutSearchTime),
+						30000,
+						0);
+	__wrap_memfault_metrics_heartbeat_set_unsigned_ExpectAndReturn(
+						MEMFAULT_METRICS_KEY(GpsSatellitesTracked),
+						2,
+						0);
+	__wrap_memfault_metrics_heartbeat_debug_trigger_Ignore();
```

Comment:
Shouldn't this be an Expect than an Ignore?

Reply:
Jepp.